urlac47192b66_ggrandes_bouncer_tgz-pJ8-org_javastack_bouncer_ClusterTestJ8/src/org/javastack/bouncer/Log.java

```diff
@@ -0,0 +1,121 @@
+package org.javastack.bouncer;
+
+import java.io.PrintStream;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * Simple logging wrapper (you want log4j/logback/slfj? easy to do!)
+ */
+public class Log {
+	public static final int LOG_NULL = 0x00;
+	public static final int LOG_CURR_STDOUT = 0x01;
+	public static final int LOG_ORIG_STDOUT = 0x02;
+
+	private static final SimpleDateFormat ISO8601DATEFORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
+	private static boolean isDebugEnabled = false;
+	private static int outMode = LOG_CURR_STDOUT;
+	private static PrintStream stdOut = System.out;
+	private static PrintStream stdErr = System.err;
+
+	static void setMode(final int newMode) {
+		outMode = newMode;
+	}
+
+	static boolean isModeOrigEnabled() {
+		return ((outMode & LOG_ORIG_STDOUT) != 0);
+	}
+
+	static boolean isModeCurrEnabled() {
+		return ((outMode & LOG_CURR_STDOUT) != 0);
+	}
+
+	static void redirStdOutLog(final String stdFile) {
+		System.setOut(new PrintStream(new AutoRotateFileOutputStream(stdFile)));
+	}
+
+	static void restoreStdOutLog() {
+		System.setOut(stdOut);
+	}
+
+	static void redirStdErrLog(final String errFile) {
+		System.setErr(new PrintStream(new AutoRotateFileOutputStream(errFile)));
+	}
+
+	static void restoreStdErrLog() {
+		System.setErr(stdErr);
+	}
+
+	static void enableDebug() {
+		isDebugEnabled = true;
+	}
+
+	static boolean isDebug() {
+		return isDebugEnabled;
+	}
+
+	static String getTimeStamp() {
+		synchronized (ISO8601DATEFORMAT) {
+			return ISO8601DATEFORMAT.format(new Date());
+		}
+	}
+
+	static void debug(final String str) {
+		if (isDebugEnabled) {
+			final String msg = getTimeStamp() + " [" + getId() + "]" + " [DEBUG] " + "[" + getName() + "] "
+					+ str;
+			if (isModeOrigEnabled())
+				stdOut.println(msg);
+			if (isModeCurrEnabled())
+				System.out.println(msg);
+		}
+	}
+
+	static void info(final String str) {
+		final String msg = getTimeStamp() + " [" + getId() + "]" + "[INFO]" + "[" + getName() + "] " + str;
+		if (isModeOrigEnabled())
+			stdOut.println(msg);
+		if (isModeCurrEnabled())
+			System.out.println(msg);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 75:79, 83:87, 91:95.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.